### DISCUSSION TITLE
SQLite Connection should also accept `Path`

### DISCUSSION THREAD

**Discussion by hasezoey** [diesel-rs/diesel#3069] [Ideas]
Because SQLite wants a path on the filesystem, there should be a way to pass a `std::path::Path` as a argument instead of just `&str` to establish a connection

i know that currently `establish` is from the trait, which probably will not be changed, but there should be a custom function that accepts it

or better it should be `AsRef<Path>`

Why? because converting a `Path` to `str` (or `String`) is lossy from what i know, because in rust strings must be UTF-8, but paths may not be

**Comment by weiznich** [comment]
I don't think that this is easily possible. Sqlite assumes that paths are either utf-8 or utf-16. 
 https://www.sqlite.org/capi3ref.html#sqlite3_open

**Reply by hasezoey** [reply]
> I don't think that this is easily possible. Sqlite assumes that paths are either utf-8 or utf-16.

now that i see the sqlite documentation, yes it seems hard to implements, and i dont quite fully understand why sqlite does not accept a non-UTF8 path (which is quite possible in some cases)
